core/class/jMQTTCmd.class.php

```diff
@@ -252,29 +252,32 @@ public function preSave() {
 		// Check "request" value if autoPub enabled
 		if ($this->getType() == 'action' && $this->getConfiguration('autoPub', 0)) {
 			$req = $this->getConfiguration('request', '');
-			// Get all commands
-			preg_match_all("/#([0-9]*)#/", $req, $matches);
-			$cmds = array_unique($matches[1]);
-			// $value = '';
-			if (count($cmds) > 0) { // There are commands
-				$this_topic = $this->isJson() ? substr($this->getTopic(), 0, strpos($this->getTopic(), '{')) : $this->getTopic();
-				foreach ($cmds as $cmd_id) {
-					$cmd = cmd::byId($cmd_id);
-					$cmd_topic = $cmd->isJson() ? substr($cmd->getTopic(), 0, strpos($cmd->getTopic(), '{')) : $cmd->getTopic();
-					if (!is_object($cmd))
-						throw new Exception('Impossible d\'activer la publication automatique sur <b>'.$this->getHumanName().'</b> car la commande <b>'.$cmd_id.'</b> est invalide.');
-					if ($cmd->getType() != 'info')
-						throw new Exception('Impossible d\'activer la publication automatique sur <b>'.$this->getHumanName().'</b> car la commande <b>'.$cmd->getHumanName().'</b> n\'est pas de type info.');
-					if ($cmd->getEqType() =='jMQTT' && $this_topic == $cmd_topic)
-						throw new Exception('Impossible d\'activer la publication automatique sur <b>'.$this->getHumanName().'</b> car la commande <b>'.$cmd->getHumanName().'</b> référence le même topic.');
-					// if ($cmd->getEqType() =='jMQTT' && $this->getEqLogic()->getBrkId() == $cmd->getEqLogic()->getBrkId())
-						// throw new Exception('Impossible d\'activer la publication automatique sur <b>'.$this->getHumanName().'</b> car la commande <b>'.$cmd->getHumanName().'</b> appartient au même Broker.');
-					// $value .= '#'.$cmd->getId().'#';
+			// If Request has changed
+			if (self::byId($this->getId())->getConfiguration('request', '') != $req) {
```

Comment:
Does this test works if we **create an action cmd** then **check autopub** and finally **save**
(because "self::byId($this->getId())" will return $null right)

Reply:
Exact, corrected in commit https://github.com/Domochip/jMQTT/pull/113/commits/7e54732eed05c2213798e64efd60d83a0b869acd